WebRoot/public/admin.jsp

```diff
@@ -68,21 +68,26 @@
 	        }
 	    }    
 	}
-	
-    Boolean isLogoff = getParameter(request, "logoff", "0").equals("1");
-    if (isLogoff) {
-        Cookie killAuthToken = new Cookie("ZM_ADMIN_AUTH_TOKEN", null);
-		killAuthToken.setMaxAge(0);
-		killAuthToken.setPath("/");
-		response.addCookie(killAuthToken);
-		String queryString = request.getQueryString();
-        queryString = queryString.replaceAll("&?logoff=1", "");
-		if (queryString.length() > 0)
-			response.sendRedirect(adminUrl + "?" + queryString);
-		else
-			response.sendRedirect(adminUrl); 
-    }	
-	
+%>
+
+<c:if test="${param.logoff eq '1'}">
+	<zm:logout isAdmin="true" />
+
+	<c:set var="req" value="${pageContext.request}" />
+	<c:set var="redirectURL" value="${req.scheme}://${req.serverName}:${req.serverPort}${req.contextPath}" />
```

Comment:
I'm not positive, but this change from using adminUrl (previous via zimbraAdminURL IIRC) to reconstructing the URL  feels off.  Consider the case where the server is behind the proxy with zimbraReverseProxyAdminEnabled set to TRUE. In that case I'm not sure the URL in the response ultimately is correct for this use case.  I'm thinking it perhaps ends up with an internal server name/port instead of the public one.  Can you verify?